### ISSUE TITLE
about resume train with yolo11

### ISSUE THREAD

**Issue by huhuhu5798** [ultralytics/ultralytics#17492]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

At first, I set my epochs to 200 and completed the training, but I want to change my epochs to 500 based on the finished training. What should I do?

**Comment by Y-T-G** [comment]
You can start training from last.pt.

https://www.reddit.com/r/Ultralytics/comments/1eolwl8/the_correct_way_to_train_from_a_previously/

But it's not "resuming", it's a new training starting from your finished model.

**Comment by purples123** [comment]
I meet the same problem. I set the `resume=True`， and  `model.load('/yolov11/runs/train/exp3/weights/last.pt')` ,but ite  just start training from 1.Is there any method to start training from last training epoch?

**Comment by AtlasCodex** [comment]
> I meet the same problem. I set the `resume=True`， and `model.load('/yolov11/runs/train/exp3/weights/last.pt')` ,but ite just start training from 1.Is there any method to start training from last training epoch?

You can try the yolo command line to continue training. I also encountered the problem you had too many times.

**Comment by purples123** [comment]
Thanks for your reply. I tried `result=model.train(resume=True)`, but it doesn't work. I solved this problem by setting resume in `trainer.py`:
![image](https://github.com/user-attachments/assets/c1e628c5-08a6-4f10-a317-f970886454b1)

thankfully it successfully continues training from where it left off.

**Comment by Y-T-G** [comment]
> I meet the same problem. I set the `resume=True`， and  `model.load('/yolov11/runs/train/exp3/weights/last.pt')` ,but ite  just start training from 1.Is there any method to start training from last training epoch?
> 

You shouldn't use `model.load`. You should load the `last.pt` using `model = YOLO("last.pt")`

**Comment by purples123** [comment]
I use the following command to train, is there anything wrong?
```python    
model = YOLO(model='/root/autodl-tmp/project/building_defect_detect/yolov11/ultralytics/cfg/models/11/yolo11.yaml')
model.load('/root/autodl-tmp/project/building_defect_detect/yolov11/runs/train/exp/weights/last.pt') 
model.train(data='data.yaml',
                imgsz=320,
                epochs=500,
                batch=360,
                workers=3,
                device=[0,1,2],
                optimizer='SGD',
                close_mosaic=10,
                resume = True,
                project='runs/train',
                name='exp',
                single_cls=False,
                cache=False,
                lr0=0.0001,
                save_period=50,
                )
```

**Comment by Y-T-G** [comment]
Yes. You're supposed to load last.pt using `YOLO()` not `model.load`

**Comment by purples123** [comment]
OK, thank you!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by OmarKhater98** [comment]
I implemented filter pruning, saved the pruned weights, and customized a new YAML file to match the new architecture. But now the evaluation metrics are always zero. Any suggestions please?

![Image](https://github.com/user-attachments/assets/4c510db4-2097-41bc-9b56-505ac7c35585)

**Comment by glenn-jocher** [comment]
For evaluation metrics showing zero after pruning, first verify your pruned YAML file correctly matches the modified architecture layers and channel counts. Next, ensure your model initialization uses the pruned weights properly via `model = YOLO('pruned.yaml').load('pruned_weights.pt')`. If metrics remain zero, your pruning might be too aggressive, removing critical layers - try gradual pruning with Ultralytics HUB Model Pruning documentation.

**Comment by OmarKhater98** [comment]
Please guide me on properly saving the model after the pruning, including the weights saved for the fine-tuning?

**Comment by Y-T-G** [comment]
If you use modified yaml, then you can just use the best.pt that was saved

**Comment by LeiaLi** [comment]
After resuming, the evaluation scores drop. Why？

**Comment by sila-a0** [comment]
Hello,  I'm training a YOLOv11 segmentation  model using Google Colab. During training, the process gets interrupted (e.g., session timeout, GPU memory issues, etc.)   I'm trying to continue with the resume parameter.  When I restart the training with resume=True using the last.pt checkpoint, the loss values become NaN (Not a Number). How can I continue in a proper way ? Thank you:)

 
 ```
from ultralytics import YOLO

# Load a model
model = YOLO("yolo11m-seg.pt")

# Train the model
train_results = model.train(
    data="/content/drive/MyDrive/data.yaml",
epochs=500,  imgsz=1280 ,batch=2,    project='/content/drive/MyDrive/SegmentationModel', save_period=50
)
```



```
from ultralytics import YOLO

# Load a model
model = YOLO("/content/drive/MyDrive/SegmentationModel/train/weights/last.pt")

# Train the model
train_results = model.train(
    resume=True,
    data="/content/drive/MyDrive/data.yaml",
epochs=500,  imgsz=1280 ,batch=2,    project='/content/drive/MyDrive/SegmentationModel', save_period=50
)

```


<img width="1207" height="142" alt="Image" src="https://github.com/user-attachments/assets/c1f52e2c-80e5-4c04-95c6-943d63027837" />

**Comment by glenn-jocher** [comment]
Hi @sila-a0, NaN losses during resume typically indicate corrupted checkpoint or learning rate issues. Try resuming without specifying additional parameters first: `model.train(resume=True)` only. If NaN persists, the checkpoint may be corrupted - use an earlier `epoch_X.pt` file instead of `last.pt`, or reduce learning rate with `lr0=0.001`. Also ensure sufficient GPU memory as Colab memory constraints can cause instability during resume.

**Comment by sila-a0** [comment]
Hello @glenn-jocher thank you very much for your advice.  model.train(resume=True)  I tried this and still NaN.


Then, I tried with epoch100.pt and still NaN. 

```
from ultralytics import YOLO


model = YOLO("/content/drive/MyDrive/SegmentationModel/train/weights/epoch100.pt")

model.train(resume=True,lr0=0.001)    
```  

<img width="839" height="175" alt="Image" src="https://github.com/user-attachments/assets/609ce1f5-4d4d-44a0-bfd4-c70380979252" />

**Comment by glenn-jocher** [comment]
Hi @sila-a0, persistent NaN losses across multiple checkpoints suggests a deeper issue. Try starting fresh training from the pretrained model (`yolo11m-seg.pt`) without resume to verify if the issue is resume-specific or general. If fresh training works normally, the checkpoints may all be corrupted. Also try reducing batch size to `batch=1` and using a smaller model like `yolo11s-seg.pt` to minimize Colab memory pressure which can cause training instability.
